MatrixSDK/MXSession.m

```diff
@@ -1672,6 +1680,9 @@ - (void)handleServerSyncError:(NSError*)error forRequestWithServerTimeout:(NSUIn
     // Check whether the caller wants to handle error himself
     if (failure)
     {
+        MXLogErrorDetails(@"[MXSession] handleServerSyncError: sync fail is handled by caller, resetting eventStreamRequest might not be handled", @{
+            @"error": error ?: @"unknown"
+        });
```

Comment:
@ismailgulek WDYT about this bit ? If the error is a timeout and failure != nil, the eventStreamRequest might not be reset 🤔

Reply:
Yeah you right. It even makes sense to reset the request here regardless of `failure` block.